---a/net/minecraft/entity/ai/EntityAIOwnerHurtByTarget.java
+++b/net/minecraft/entity/ai/EntityAIOwnerHurtByTarget.java
@@ -3,35 +3,29 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.passive.EntityTameable;
 
-public class EntityAIOwnerHurtByTarget extends EntityAITarget
-{
+public class EntityAIOwnerHurtByTarget extends EntityAITarget {
+
     EntityTameable theDefendingTameable;
     EntityLivingBase theOwnerAttacker;
     private int field_142051_e;
     private static final String __OBFID = "CL_00001624";
 
-    public EntityAIOwnerHurtByTarget(EntityTameable p_i1667_1_)
-    {
+    public EntityAIOwnerHurtByTarget(EntityTameable p_i1667_1_) {
         super(p_i1667_1_, false);
         this.theDefendingTameable = p_i1667_1_;
         this.setMutexBits(1);
     }
 
-    public boolean shouldExecute()
-    {
-        if (!this.theDefendingTameable.isTamed())
-        {
+    public boolean shouldExecute() {
+        if (!this.theDefendingTameable.isTamed()) {
             return false;
-        }
-        else
-        {
+        } else {
             EntityLivingBase entitylivingbase = this.theDefendingTameable.getOwnerEntity();
 
-            if (entitylivingbase == null)
-            {
+            if (entitylivingbase == null) {
                 return false;
-            }
-            else
-            {
+            } else {
                 this.theOwnerAttacker = entitylivingbase.getAITarget();
                 int i = entitylivingbase.getRevengeTimer();
                 return i != this.field_142051_e && this.isSuitableTarget(this.theOwnerAttacker, false) && this.theDefendingTameable.func_142018_a(this.theOwnerAttacker, entitylivingbase);
@@ -40,18 +40,15 @@
         }
     }
 
-    public void startExecuting()
-    {
-        this.taskOwner.setAttackTarget(this.theOwnerAttacker);
+    public void startExecuting() {
+        this.taskOwner.setGoalTarget(this.theOwnerAttacker, org.bukkit.event.entity.EntityTargetEvent.TargetReason.TARGET_ATTACKED_OWNER, true); // CraftBukkit - reason
         EntityLivingBase entitylivingbase = this.theDefendingTameable.getOwnerEntity();
 
-        if (entitylivingbase != null)
-        {
+        if (entitylivingbase != null) {
             this.field_142051_e = entitylivingbase.getRevengeTimer();
         }
 
         super.startExecuting();
     }
-}
+}
